man/man8/zpool-events.8

```diff
@@ -66,8 +66,431 @@ single tab instead of arbitrary space.
 Print the entire payload for each event.
 .El
 .
+.Sh EVENTS
+Most of these don't have any description.
+The events generated by ZFS have never been publicly documented.
+What is here is intended as a starting point to provide documentation
+for all possible events.
+.Pp
+To view all events created since the loading of the ZFS infrastructure
+(i.e. "the module"), run
+.Dl Nm zpool Cm events
+to get a short list, and
+.Dl Nm zpool Cm events Fl v
+to get a full detail of the events and what information
+is available about it.
+.Pp
+This manual page lists the different subclasses that are issued
+in the case of an event.
+The full event name would be
+.Sy ereport.fs.zfs.\& Ns Em SUBCLASS ,
+but we only list the last part here.
```

Comment:
Most of the existing EVENTS text here was a bit redundant (or out of date) so why don't we go ahead and drop the first two paragraphs, then reword the last paragraph and move it under the EVENTS (SUBCLASSES) section.  So then there wouldn't be any EVENTS section.  Here's a snippet of what I was thinking.

```
     ...
     -v      Print the entire payload for each event.

EVENTS (SUBCLASSES)
     The following is a list of the different event subclasses.  The full
     event name would be ereport.fs.zfs.SUBCLASS, but we only list the last
     part here.

     checksum           Issued when a checksum error has been detected.
     ...
```

Reply:
How about dropping the `(SUBCLASSES)` from the sexion heading?
It makes little sense to the first-time reader, and they them being a "subclass" is both explained away immediately, and (since the prefix is constant) entirely irrelevant – like this:
```
     -v      Print the entire payload for each event.

EVENTS
     These are the different event subclasses.  The full event name would be
     ereport.fs.zfs.SUBCLASS, but only the last part is listed here.

     checksum           Issued when a checksum error has been detected.
```

Reply:
Good point, that works for me.

Reply:
Cool, updated.